tsconfig.esm.json

```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "outDir": "./esm",
     "module": "es2022",
+    "moduleResolution": "node",
```

Comment:
locally the esm tsc build failed without this. i'm not sure how it ever worked without it 🤔

Reply:
We have been using the default and have not had issues, could you pull this out of this PR?